src/libexpr/primops.cc

```diff
@@ -4074,37 +4074,150 @@ void EvalState::createBaseEnv()
 
     /* `builtins' must be first! */
     v.mkAttrs(buildBindings(128).finish());
-    addConstant("builtins", v);
+    addConstant("builtins", v, {
+        .type = "attribute set",
+        .doc = R"(
+          Contains all the [built-in functions](./builtins.md) and values, in order to avoid polluting the global scope.
+
+          Since built-in functions were added over time, [testing for attributes](./operators.md#has-attribute) in `builtins` can be used for graceful fallback on older Nix installations:
+
+          ```nix
+          if builtins ? getEnv then builtins.getEnv "PATH" else ""
+          ```
+        )",
+    });
 
     v.mkBool(true);
-    addConstant("true", v);
+    addConstant("true", v, {
+        .type = "boolean",
+        .doc = R"(
+          The boolean value for truth.
+
+          Users familiar with other languages might assume `true` is special syntax in the Nix language,
+          but this is in fact not the case:
+
+          ```nix-repl
+          nix-repl> let true = 1; in true
+          1
+          ```
```

Comment:
I'm very much against this narrative style in a reference manual. It's completely irrelevant what users may or may not be familiar with. What's more important but not obvious is how that interacts with conditionals. Also, since it's just a primitive value, strictly speaking without further information it has no semantics whatsoever.

````suggestion
          Primitive value.

          It can be returned by [comparison operators](...) and used in [conditional expressions](...).

          The name `true` is not special, and can be shadowed:

          ```nix-repl
          nix-repl> let true = 1; in true
          1
          ```
````